src/cls/IPM/ResourceProcessor/WebApplication.cls

```diff
@@ -216,15 +216,23 @@ ClassMethod DoPropertiesExist(pPropList As %Library.List) As %Library.List [ Int
 
 Method ConstructRoleName(pDbResource As %String) As %String [ Internal, Private ]
 {
-    // Include the module name in the role name to avoid conflicts
+    // Include the module name in the role name to avoid conflicts, using hash because role name can't be longer than 64 characters
     // This role should only be managed by IPM itself
-    Return $ListToString($lb("IPM", "readonly", ..ResourceReference.Module.Name, pDbResource), ".")
+    Set hashBytes = $System.Encryption.MD5Hash(..ResourceReference.Module.Name) // 128 bits
+    Set hashHexString = $System.Encryption.Base32HexEncode(hashBytes, 1) // 26 characters, without '=' padding
+    // As long as pDbResource is shorter than 33 characters, the role name will be shorter than 64 characters
+    Set roleName = $ListToString($ListBuild("IPM", pDbResource, hashHexString), ".")
```

Comment:
Nit: The resource could begin with %DB_ so this could look a bit odd. Also in general resource names are `-` or `_` separated so I think the following would be more appropriate:
```
$ListToString($ListBuild(pDbResource, "IPM", hashHexString), "_")
```